### PR TITLE
fix-discriminator-mapper generation issue

### PR DESCRIPTION
**Pull request by qiaozha**
fix https://github.com/Azure/autorest.typescript/issues/1395

### CODE REVIEW

**Review by joheredi — APPROVED** [review]
Looks good to me @sarangan12 can you also take a look please?

File: packages/autorest.typescript/test/smoke/generated/graphrbac-data-plane/review/graphrbac-data-plane.api.md [hunk 1]
```diff
@@ -283,7 +283,7 @@ export interface DirectoryObject {
     [property: string]: any;
     readonly deletionTimestamp?: Date;
     readonly objectId?: string;
-    objectType: "User" | "Application" | "Group" | "ServicePrincipal";
+    objectType: string;
```
Comment (qiaozha) [inline packages/autorest.typescript/test/smoke/generated/graphrbac-data-plane/review/graphrbac-data-plane.api.md:286]: This is expected. DirectoryObject has four subclasses, "User" , "Application",  "Group" , "ServicePrincipal", but they didn't specify x-ms-discriminator-value, in this case the discriminator value for this four subclasses should have the value of the  class name itself. but that doesn't mean objectType can only have this four values.

File: packages/autorest.typescript/test/smoke/generated/graphrbac-data-plane/src/models/mappers.ts [hunk 2]
```diff
@@ -2282,7 +2282,7 @@ export const UserUpdateParameters: coreClient.CompositeMapper = {
 };
 
 export let discriminators = {
-  DirectoryObject: DirectoryObject,
+  "DirectoryObject.DirectoryObject": DirectoryObject,
```
Comment (qiaozha) [inline packages/autorest.typescript/test/smoke/generated/graphrbac-data-plane/src/models/mappers.ts:2285]: This may be unexpected.

**Review by sarangan12 — APPROVED** [review]

File: packages/autorest.typescript/test/smoke/generated/graphrbac-data-plane/src/models/mappers.ts (hunk 2 above)
Comment (qiaozha) [reply packages/autorest.typescript/test/smoke/generated/graphrbac-data-plane/src/models/mappers.ts]: fixed

**Pull request merged**